SonicWall/sonicwall-fw/tests/test_app_control_detection_alert.json

```diff
@@ -0,0 +1,83 @@
+      "intake": {
+        "dialect": "SonicWall Firewall",
+        "dialect_uuid": "ee0b3023-524c-40f6-baf5-b69c7b679887"
+      }
+    },
+    "message": "CEF:0|SonicWall|NSa 4700|7.0.1-1234-R5678|1154|Application Control Detection Alert|9|cat=0 gcat=3 smac=00:11:22:33:44:55 src=10.0.10.20 spt= 12345 deviceInboundInterface=X1 cs3Label=WAN dmac=1a:c2:41:00:65:11 dst=10.0.20.30 dpt=49773 deviceOutboundInterface=X6-V320 cs4Label=WORKSTATION proto=tcp/https in=1240 app=49000 appName=\"General HTTPS\" sid=7900 appcat=\"PROTOCOLS\" appid=1234 catid=77 msg=\"Application Control Detection Alert: PROTOCOLS SSL/TLS Protocol -- TLSv1.2 Version, SID: 7900, AppID: 1200, CatID: 77\" msg=\"Application Control Detection Alert: PROTOCOLS SSL/TLS Protocol -- TLSv1.2 Version\" sid=7800 appcat=\"PROTOCOLS SSL/TLS Protocol -- TLSv1.2 Version\" appid=1234 catid=55 cnt=7800123 fw_action=\"NA\""
+  },
+  "expected": {
+    "message": "CEF:0|SonicWall|NSa 4700|7.0.1-1234-R5678|1154|Application Control Detection Alert|9|cat=0 gcat=3 smac=00:11:22:33:44:55 src=10.0.10.20 spt= 12345 deviceInboundInterface=X1 cs3Label=WAN dmac=1a:c2:41:00:65:11 dst=10.0.20.30 dpt=49773 deviceOutboundInterface=X6-V320 cs4Label=WORKSTATION proto=tcp/https in=1240 app=49000 appName=\"General HTTPS\" sid=7900 appcat=\"PROTOCOLS\" appid=1234 catid=77 msg=\"Application Control Detection Alert: PROTOCOLS SSL/TLS Protocol -- TLSv1.2 Version, SID: 7900, AppID: 1200, CatID: 77\" msg=\"Application Control Detection Alert: PROTOCOLS SSL/TLS Protocol -- TLSv1.2 Version\" sid=7800 appcat=\"PROTOCOLS SSL/TLS Protocol -- TLSv1.2 Version\" appid=1234 catid=55 cnt=7800123 fw_action=\"NA\"",
+    "event": {
+      "severity": 9,
+      "reason": "Application Control Detection Alert: PROTOCOLS SSL/TLS Protocol -- TLSv1.2 Version, SID: 7900, AppID: 1200, CatID: 77"
+    },
+    "observer": {
+      "vendor": "SonicWall",
+      "type": "firewall",
+      "version": "7.0.1-1234-R5678",
+      "egress": {
+        "interface": {
+          "name": "X6-V320"
+        }
+      },
+      "ingress": {
+        "interface": {
+          "name": "X1"
+        }
+      }
+    },
+    "rule": {
+      "id": "1154"
+    },
+    "source": {
+      "ip": "10.0.10.20",
+      "mac": "00:11:22:33:44:55",
+      "port": 12345,
+      "address": "10.0.10.20"
+    },
+    "destination": {
+      "port": 49773,
+      "ip": "10.0.20.30",
+      "mac": "1a:c2:41:00:65:11",
+      "bytes": 1240,
+      "address": "10.0.20.30"
+    },
+    "host": {
+      "network": {
+        "ingress": {
+          "bytes": 1240
+        }
+      }
+    },
+    "cef": {},
+    "network": {
+      "transport": "tcp/https"
+    },
+    "process": {
+      "entity_id": "1234",
+      "name": "General HTTPS"
+    },
+    "sonicwall": {
+      "fw": {
+        "appcat": "PROTOCOLS",
+        "cat": "0",
+        "catid": "77",
+        "cnt": "7800123",
+        "cs3Label": "WAN",
+        "cs4Label": "WORKSTATION",
+        "fw_action": "NA",
```

Comment:
`NA` values should be discarded from the result. Please update the parser :pray: